src/main/kotlin/it/pagopa/wallet/services/PaymentMethodsService.kt

```diff
@@ -0,0 +1,49 @@
+package it.pagopa.wallet.services
+
+import it.pagopa.generated.ecommerce.model.PaymentMethodResponse
+import it.pagopa.wallet.client.EcommercePaymentMethodsClient
+import it.pagopa.wallet.repositories.PaymentMethodsTemplateWrapper
+import kotlinx.coroutines.reactor.mono
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+import reactor.kotlin.core.publisher.switchIfEmpty
+
+@Service
+class PaymentMethodsService(
+    @Autowired private val paymentMethodsRedisTemplate: PaymentMethodsTemplateWrapper,
+    @Autowired private val ecommercePaymentMethodsClient: EcommercePaymentMethodsClient,
+) {
+
+    /*
+     * Logger instance
+     */
+    var logger: Logger = LoggerFactory.getLogger(this.javaClass)
+
+    fun getPaymentMethodById(paymentMethodId: String): Mono<PaymentMethodResponse> =
```

Comment:
What happens if we fail to retrieve the payments methods from both cache and api call? Should we propagate an error?

Reply:
EcommercePaymentMethodException is propagated when fail api call. This logic is the same as before.